src/styles/common/StyledButton.tsx

```diff
@@ -195,22 +204,41 @@ export const ButtonTheme = variants<'kind', keyof typeof ButtonVariations>('mode
   },
 })
 
-/*
-TODO: consider adding:
-  &.small {
-    font-size: 0.6rem;
-    padding: 0.3rem 0.5rem;
-  }
-
-  &.big {
-    font-size: 1.2rem;
-    padding: 0.65rem 1rem;
-  }
-*/
+// Created a 'size' prop on buttons, default | small | big
+const ButtonSizes = variants('component', 'size', {
+  default: {
+    buttons: '',
+  },
+  small: {
+    buttons: css`
+      font-size: 0.6rem;
+      padding: 0.3rem 0.5rem;
+    `,
+  },
+  big: {
+    buttons: css`
+      font-size: 1.4rem;
+      padding: 0.65rem 1rem;
+    `,
+  },
+})
 
-export default styled.button`
+const ColouredButtonBase = styled.button`
   border: ${buttonBorder};
-
   /* Fold in theme css above */
   ${ButtonTheme}
 `
+
+const ColouredAndSizedButtonBase = styled(ColouredButtonBase)`
+  font-size: ${buttonFontSize};
+  ${ButtonSizes}
+`
+// Wrap ColouredAndSizedButtonsBase in it's own ThemeProvider which takes the toplevel app theme
+// ThemeProvider and interpolate over it's props
+const ThemedButtonBase: React.FC<React.ButtonHTMLAttributes<Element>> = (props) => (
+  <ThemeProvider theme={({ mode }: ThemeMode): ThemeMap => ({ mode, component: 'buttons' })}>
```

Comment:
I'm not entirely sure how `styled-theming` works, but can't we, instead of an extra Provider, do `variants('mode', 'size'`, like [here](https://jamie.build/styled-theming.html#example:~:text=Multi%2Ddimensional%20theming)

Reply:
yes but the theme structure then falls on sth like:
```js
{ 
  big: {
    light: { ... }
  }
}
```

where `light` and `big` have no connection